#9 로그인 aop 적용

0hun opened the pull request:
aop 의존성 추가
LoginCheck annotion으로 aop 적용

0hun committed:
#9 로그인 aop 적용

aop 의존성 추가
LoginCheck annotion으로 aop 적용

f-lab-dev commented on src/main/java/com/younghun/hibusgo/aop/LoginCheckAop.java:
```diff
@@ -0,0 +1,30 @@
+    boolean islogin = loginService.isLoginAccount();
+
+    if (!islogin) {
+      throw new HttpStatusCodeException(HttpStatus.UNAUTHORIZED){};
```
뒤에 `{}`는 오타인가요?

0hun replied:
구현을 해야하는 추상 메소드이기 때문에 내부를 구현하지 않는 의미로 사용하였습니다.
현재는 추상클래스를 상속한 HttpClientErrorException를 사용하도록 변경하였습니다.

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/AccountController.java:
```diff
@@ -100,14 +97,11 @@ public ResponseEntity deleteAccount() {
+    @LoginCheck
     @PatchMapping("/password")
     public ResponseEntity updatePassword(@RequestBody @Valid PasswordDto passwordDto, Errors errors) {
         String sessionId = loginService.getLoginAccountId();
```
잘 적용하셨는데 이렇게 로그인한 유저의 id도 메소드 인자로 주입받을 수 있게 해주면 어떨까요?

0hun replied:
넵 수정 반영하겠습니다.

0hun committed:
#9 로그인 aop 적용

 loginId가 필요한 경우 PathVariable를 통해 받도록 수정

 HttpStatusCodeException는 추상 클래스이기 때문에 내부를
생성자가 protected로 선언되있어 이를 상속하여 사용하는 HttpClientErrorException로 변경

ProfileDto로 id,password를 제외한 프로필에 관한
정보만 받아서 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/AccountController.java:
```diff
@@ -78,11 +80,9 @@ public ResponseEntity addAccount(@RequestBody @Valid AccountDto accountDto, Erro
-
-        accountService.deleteAccount(sessionId);
+    @DeleteMapping("/myInfo/{loginId}")
+    public ResponseEntity deleteAccount(@PathVariable String loginId) {
```
자신이 탈퇴하는 것이기 때문에 이렇게 외부에서 입력받는게 아니여야합니다.
이 `@PathVariable`처럼 우리가 뭔가 어노테이션을 하나 더 만들어서 인자로 **세션에서 불러온 유저의 아이디**를 주입해줄 수 있지 않을까요?

왜냐면 모든 로그인이 필요한 API에다가 유저의 ID를 입력하라고 요구할 순 없으니까요~

0hun replied:
넵 알겠습니다.

f-lab-dev commented on src/main/java/com/younghun/hibusgo/dto/ProfileDto.java:
```diff
@@ -0,0 +1,37 @@
+  private String name;
+
+  // 이메일
+  @Email
```
API의 스펙을 잘 알아보고 사용하시는 것 같아서 보기 좋네요 👍

0hun committed:
#9 로그인 aop 적용

 리뷰 반영

 세션에서 불러온 유저의 아이디를 특정 anntation에만 적용하도록 수정
 annotaion에 해당하는 파라미터에 유저의 세션 아이디를 바인딩 하도록
 HandlerMethodArgumentResolver 설정 및 등록

f-lab-dev commented on src/main/java/com/younghun/hibusgo/utils/SessionIdResolver.java:
```diff
@@ -0,0 +1,31 @@
+
+      String loginId = loginService.getLoginAccountId();
+
+    return loginId;
```
여기 들여쓰기 오타같네요~

0hun replied:
넵 수정하겠습니다,

f-lab-dev reviewed (approved):
보통은 직접 AOP를 이용해 구현하는데 더 좋은 방법으로 구현하셨네요~
스프링의 스펙을 꼼꼼하게 잘 알고 사용하시는 것 같아서 좋습니다. 코드도 점점 깔끔해지고요.
그냥 reflection 공부용으로 aspect를 직접 개발하는 것도 한번 해보시는것도 좋을 것 같네요~

리뷰 1개 해결하시고 머지하시면 될 것 같습니다~
오늘 10시에 뵙겠습니다

0hun committed:
Merge branch 'develop' into issue/9
